RELEASE/scripts/autoscend/quests/level_12.ash

```diff
@@ -799,14 +818,24 @@ boolean L12_filthworms()
 		use(1, $item[Filthworm Royal Guard Scent Gland]);
 	}
 
+	//if we can kill the queen we don't care about gland drops anymore. kill her and finish this
 	if(have_effect($effect[Filthworm Guard Stench]) > 0)
 	{
-		handleFamiliar("meat");
-		autoAdv(1, $location[The Filthworm Queen\'s Chamber]);
-		return true;
+		return autoAdv(1, $location[The Filthworm Queen\'s Chamber]);
 	}
 
-	if(!have_skill($skill[Lash of the Cobra]))
+	//if we can guarentee stealing the stench gland then no point in buffing item drop
+	if(auto_have_skill($skill[Lash of the Cobra]) && get_property("_edLashCount").to_int() < 31)
+	{
+		auto_log_info("Ed will steal stench glands using [Lash of the Cobra]");
+	}
+	else if(get_property("_xoHugsUsed").to_int() < 11 && canChangeToFamiliar($familiar[XO Skeleton]))
+	{
+		auto_log_info("Will steal stench glands using [XO Skeleton]");
+		handleFamiliar($familiar[XO Skeleton]);
+	}
+	//TODO add IOTM cat burglar stealing support here with another else if
```

Comment:
If we ever get around to adding support for Zombie Master there's a skill called [Smash & Graaagh](https://kol.coldfront.net/thekolwiki/index.php/Smash_%26_Graaagh) which works the same as the previous 2 (ignore for now though just mentioning if you want to add a TODO).

Reply:
added it... then commented it all out. 
because it seems like we are missing some mafia side tracking on this. We have zero support for that path so far. we need to add to auto_combat using their steal in all the places we use lash, and we can't really do it without proper mafia tracking (well, I guess we could track it ourselves but better mafia track it for us)

Reply:
ZM might predate mafia then. I don't know the history well enough but you're right we shouldn't care about stuff mafia doesn't support (and note I said to ignore unless you wanted to add a TODO comment for it).